Сделал вторую лабу. Было больно.

CandyJunkie opened the pull request:
Для тех кто читает, не повторяйте моих ошибок и не считайте дисперсию. :)

CandyJunkie committed:
Merge pull request #3 from dbondin/master

Обновляем наш форк

CandyJunkie committed:
Добавил вторую лабу.

Т.к. я ленивая жопа, makefile делать лень. Прилагаю проектный файл.

CandyJunkie committed:
Добавил оценку за вторую лабу

CandyJunkie commented:
Для тех кто читает, не повторяйте моих ошибок и не считайте дисперсию. :)

CandyJunkie committed:
Поправил комменты

dbondin commented on labs/lab2/CandyJunkie/lab2.cpp:
```diff
@@ -0,0 +1,271 @@
+{
+	/////------------------------------| Подготовочная часть |------------------------------/////
+
+	//system ("chcp 1251"); // включаем для отладки в винде.
```
#ifdef _WIN32
system("chcp 1251")
#endif // _WIN32 И не надо никаких туда-сюда коментариев

dbondin commented on labs/lab2/CandyJunkie/lab2.cpp:
```diff
@@ -0,0 +1,271 @@
+	unsigned char mfs = most_frequent_symbol (input);
+	if ('о' == koi8[mfs]) {
+		//std::cout << 1 << ' ' << mfs << ";\n";
+		return 1;
```
Назачем int? Ну почему не сделать enum с нормальными говорящими названиями. 
Ведь потом фиг поймешь, что такое 1 и что такое 2

dbondin commented on labs/lab2/CandyJunkie/lab2.cpp:
```diff
@@ -0,0 +1,271 @@
+
+	//system ("chcp 1251"); // включаем для отладки в винде.
+
+	for (int i = 0; i < 256; ++i) {
```
Правда-же всю эту ерунду с заполнением табличек можно сделать один раз во-вспомогательной программке, которая все заполнит и сгенерит какой-нибудь codes.h со всеми этими массивами уже заполненными.
Зачем при каждом запуске это все колдовать?

dbondin commented on labs/lab2/CandyJunkie/lab2.cpp:
```diff
@@ -0,0 +1,271 @@
+	//std::cout << "done\n";
+
+	char c;
+	while (text.get (c)) {
```
Эээээ.... весь файл читаем в std::string ? Целиком?
А если это полное собрание сочинений графа Толстого?

dbondin reviewed (changes requested):
Мне не нравится, что весь файл целиком читается в память. Так нельзя.

Остальные каменты - не принципиальные

CandyJunkie committed:
Изменил файлик с лабой

Теперь читаем в память только первые 1000 русских символов. Отдельный .h сделал, он генерируется, если его еще нет, но тогда сама программа не исполняется.

CandyJunkie committed:
Добавил header

Добавил, чтобы не нужно было генерировать.

CandyJunkie commented:
По всем пунктам исправил.

CandyJunkie committed:
Закомментировал отладочную информацию

CandyJunkie committed:
Проектные файлы обновил

Не помню, нужно ли обновлять, если добавился header. На всякий случай.
